variables.tf

```diff
@@ -201,3 +201,25 @@ variable "metrics" {
   }
   default = []
 }
+
+variable "mirroring_enabled" {
+  type        = bool
+  description = "Set this to true to enable mirroring. Mirroring enables messages in one Event Streams service instance to be continuously copied to a second instance to increase resiliency. See https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-mirroring."
+  default     = false
+}
+
+variable "mirroring_topic_patterns" {
+  type        = list(string)
+  description = "The list of the topics to set in instance. Required only if var.mirroring_enabled is set to true."
+  default     = null
+}
+
+variable "mirroring" {
```

Comment:
the variable mirroring needs to have the full mirroring configuration. A user has an option to enable/disbale topic/group renaming. For reference of a full mirroring configuration options see the https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/327#issuecomment-2444438593

Reply:
@srikant-sahu, I am not sure how that is going to work because if `type` is changed to any of the values like rename, use_alias or none, configuration has to be set accordingly. For ex.

if type is `none`, then mirror config looks like below

```
mirroring = {
       source_crn   = data.ibm_resource_instance.es_instance_source.id
       source_alias = "source-alias"
       target_alias = "target-alias"
       options = {
         topic_name_transform = {
            type = "none" 
            none = {
                <values to be set>
            }
         }
         group_id_transform = {
           type = "none" 
            none = {
                <values to be set>
            }
         }
       }
     }
   }
```
if type is `use_alias`, then mirror config looks like below
```
mirroring = {
       source_crn   = data.ibm_resource_instance.es_instance_source.id
       source_alias = "source-alias"
       target_alias = "target-alias"
       options = {
         topic_name_transform = {
            type = "use_alias" 
            use_alias = {
                <values to be set>
            }
         }
         group_id_transform = {
           type = "use_alias" 
            use_alias = {
                <values to be set>
            }
         }
       }
     }
   }

```

if type is `rename`, then mirror config looks like below
```
mirroring = {
       source_crn   = data.ibm_resource_instance.es_instance_source.id
       source_alias = "source-alias"
       target_alias = "target-alias"
       options = {
         topic_name_transform = {
            type = "rename" 
            rename = {
              add_prefix = "add_prefix"
              add_suffix = "add_suffix"
              remove_prefix = "remove_prefix"
              remove_suffix = "remove_suffix"
            }
         }
         group_id_transform = {
           type = "rename" 
           rename = {
             add_prefix = "add_prefix"
             add_suffix = "add_suffix"
             remove_prefix = "remove_prefix"
             remove_suffix = "remove_suffix"
           }
         }
       }
     }
```
@ocofaigh/@JunliWang, what do you suggest?

Reply:
adding and removing prefix and suffix is only required if type is `rename` for topic and consumer groups. You can see examples in the [doc](https://cloud.ibm.com/docs/EventStreams?topic=EventStreams-mirroring_setup#step5_selecttopics)

Reply:
updated the `mirroring` variable.